packages/generator/src/action-function-import/request-builder-arguments.ts

```diff
@@ -9,13 +10,32 @@ export function getRequestBuilderArgumentsBase(
   actionFunctionImport: VdmFunctionImport | VdmActionImport,
   service: VdmServiceMetadata
 ): string[] {
+  const transformer = getTransformer(actionFunctionImport, service);
   return [
     `'${service.servicePath}'`,
     `'${actionFunctionImport.originalName}'`,
-    `(data) => ${responseTransformerFunctionName(
-      actionFunctionImport.returnType,
-      service.oDataVersion
-    )}(data, ${actionFunctionImport.returnType.builderFunction})`,
+    transformer,
     'params'
   ];
 }
+
+function getTransformer(
+  actionFunctionImport: VdmFunctionImport | VdmActionImport,
+  service: VdmServiceMetadata
+): string {
+  if (
+    actionFunctionImport.returnType.returnTypeCategory ===
+    VdmReturnTypeCategory.ENTITY_NOT_DESERIALIZABLE
+  ) {
+    return `(data) => throwErrorWhenReturnTypeIsUnionType(data, '${actionFunctionImport.originalName}')`;
+  }
+  if (actionFunctionImport.returnType.builderFunction) {
```

Comment:
This check is only here because we had to make the `builderFunction` optional. The error before can actually never happen, right?

If we would make the `actionFunctionImport.returnType` optional we might be able to avoid this. What do you think?